core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/DatasourceServiceTest.java

```diff
@@ -189,4 +194,29 @@ public boolean matches( final Object arg ) {
       return true;
     }
   }
+
+  @Test
+  public void testGetDatasourceLoadThreadCount() throws IllegalArgumentException {
```

Comment:
Still need a test that actually verifies the the code:
```
String threadCountAsString =
        pluginResourceLoader.getPluginSetting( getClass(), "settings/data-access-datasource-load-threads" );
```
is actually retrieving the value in the settings.xml file.

Reply:
You shouldn't have to do any mocking. You should be able to supply an settings.xml in the test resources.
Here is how other code initilized the PentahoSystem in test code.

[https://github.com/pentaho/pentaho-kettle/blob/master/plugins/pur/core/src/test/java/org/pentaho/di/repository/utils/RepositoryFactoryTest.java#[…]2](https://github.com/pentaho/pentaho-kettle/blob/master/plugins/pur/core/src/test/java/org/pentaho/di/repository/utils/RepositoryFactoryTest.java#L48-L52)
https://github.com/pentaho/pentaho-metaverse/blob/master/core/src/it/java/org/pentaho/metaverse/analyzer/kettle/ExternalResourceConsumerIT.java#L111

Reply:
I created below JIRA case for writing the integration test for the getDatasourceLoadThreadCount().
[https://jira.pentaho.com/browse/BACKLOG-35965]